.github/workflows/check_format.yml

```diff
@@ -0,0 +1,40 @@
+on:
+  push:
+  pull_request:
+
+concurrency:
+  group: 'ci-${{ github.workflow }}-${{ github.ref }}'
+  cancel-in-progress: true
+defaults:
+  run:
+    shell: sh -ex {0}
+
+jobs:
+  check_format:
+    strategy:
+      fail-fast: false
+      matrix:
+        env: [flake8, mypy, pylint, black, isort]
```

Comment:
I wonder if it's really worth having one test matrix element per TESTENV, instead of doing something like `tox -e flake8 mypy pylint black isort` or `tox --parallel -e ...` or similar via `TOXENV`.

We may want to run some environments on different Ubuntu releases (e.g. flake8 or pylint), to test them with different Python versions, and the test matrix easily starts growing. AIUI the GitHub enterprise account should allow 180 concurrent jobs (https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration), which is not little but also not infinite, so maybe we should use less jobs when it makes sense?

(I see you just replicated what we have in Travis, but maybe it's the right time to discuss this.)

Reply:
If I understand our recent discussion about this correctly, it sounded like (please correct me if I'm wrong), we might want to hold off on increasing this for a) more information about number of jobs and tests concurrently on the enterprise account and b) potential for self-hosted test runners providing this coverage.

Reply:
Yes that's it, and again I tried to ask about the status of the github.com/canonical account but apparently I don't know *who* to ask as I've got no answer.

What do you see as the main advantage of having each environment in a separate job? Is it speed (everything parallel) or "failure clarity" (it's immediate to see which job failed, and therefore which environment failed)? If it's the latter, then a middle ground solution could be calling environments in separate steps, e.g.
```
- name: Tox: flake8
  run: tox -e pyenv
- name: Tox: black
  run: tox -e black
- ...
```
This keeps things pretty clear in the workflow logs, but clearly it will be slightly slower.

Reply:
> Yes that's it, and again I tried to ask about the status of the github.com/canonical account but apparently I don't know who to ask as I've got no answer.

Thanks for trying :)

Comments in the `.travis.yml` imply that it was optimized for fast failure reporting (tests run in parallel and report early) as well as for total time (comment implies that ordering integration tests before others means it starts earlier). These are defaults that I personally like, but I don't feel strongly about this if we think a different optimum would be better.

"Waiting for tests to finish" is usually something which can be done asynchronously, however serializing the tests would make the wait several minutes longer so I think more feedback from regular contributors is probably warranted if we're considering that. @blackboxsw @TheRealFalcon @aciba90 or others, any objections, preferences. or concerns on this?

Reply:
Most of the formatting checkers take <5 seconds, so I think it's fine to run them serially. I'd put pylint at the end as that usually takes the longest. If/When we start introducing the unit tests and integration tests, I would probably start running those in parallel as they can take longer to run.

I'm not strongly opinionated at this point though, so I'm +1 either way.

Reply:
To @paride's concern, given that it's just a single yaml file to update, I'm not really concerned about runner availability until it actually becomes a problem. If it does become a problem, it should be a fairly quick fix to run more things serially as needed.